fetcher-pipeline/app.py

```diff
@@ -74,17 +82,143 @@ def get_report(reports):
             return report
 
         return None
+    
+    def send_messages_to_queue(self, exclude_args:List):
+        """ Send messages to queue """
+        logging.debug("Sending messages to queue")
+        try: 
+            cdn_url = os.getenv('CDN_BASE_URL')
+            bus_messages = []
+            def chunk_array(in_array, size):
+                return [in_array[i:i + size] for i in range(0, len(in_array), size)]
+
+            with open("./book_catalog.json", "r") as json_file:
+                books = json.load(json_file)
+
 
+            # iterate through each language
+            logging.debug(f"iterating through {self.__ftp_dir}")
+            for language_dir in self.__ftp_dir.iterdir():
+                if language_dir.is_dir() or "analysis" not in language_dir:
```

Comment:
Generally, when you have the situation where you're in a loop and you only want to execute if certain things are true, it's more readable to "fail-fast": test for the negative and skip the loop if it doesn't match.  It also allows you to keep your indenting much shallower.

So in this case:

```python
for language...:

    if not (langauge_dir.is_dir() and "analysis" not in language_dir):
        continue 

    logging.debug(...
```